src/lib/features/onboarding/onboarding-store.ts

```diff
@@ -0,0 +1,136 @@
+import type { Db } from '../../db/db';
+import type {
+    IOnboardingStore,
+    OnboardingEvent,
+    SharedEvent,
+} from './onboarding-store-type';
+import { millisecondsToSeconds } from 'date-fns';
+
+type DBInstanceType = {
+    event:
+        | 'first-user-login'
+        | 'second-user-login'
+        | 'first-flag'
+        | 'first-pre-live'
+        | 'first-live';
+    time_to_event: number;
+};
+
+type DBProjectType = {
+    event: 'firstFlag' | 'firstPreLive' | 'firstLive';
+    time_to_event: number;
+    project: string;
+};
+
+const translateEvent = (
+    event: OnboardingEvent['type'],
+): DBInstanceType['event'] => {
+    if (event === 'flag-created') {
+        return 'first-flag';
```

Comment:
can be also first-flag-created.
Overall progression: 'initial' in lifecycle -> 'flag-created' in service -> 'first-flag-created' when we confirm it's the first one